src/Pim/Component/Catalog/Completeness/Checker/ChainedProductValueCompleteChecker.php

```diff
@@ -21,32 +31,43 @@ class ChainedProductValueCompleteChecker implements ProductValueCompleteCheckerI
      */
     public function isComplete(
         ProductValueInterface $productValue,
-        ChannelInterface $channel = null,
-        LocaleInterface $locale = null
+        ChannelInterface $channel,
+        LocaleInterface $locale
     ) {
         foreach ($this->productValueCheckers as $productValueChecker) {
-            if ($productValueChecker->supportsValue($productValue)
-                && !$productValueChecker->isComplete($productValue, $channel, $locale)
+            if ($productValueChecker->supportsValue($productValue, $channel, $locale)
+                && $productValueChecker->isComplete($productValue, $channel, $locale)
             ) {
-                return false;
+                return true;
             }
         }
 
-        return true;
+        return false;
     }
 
     /**
      * {@inheritdoc}
      */
-    public function supportsValue(ProductValueInterface $productValue)
-    {
-        foreach ($this->productValueCheckers as $productValueChecker) {
-            if ($productValueChecker->supportsValue($productValue)) {
-                return true;
-            }
+    public function supportsValue(
+        ProductValueInterface $productValue,
+        ChannelInterface $channel,
+        LocaleInterface $locale
+    ) {
+        if (null !== $productValue->getScope() && $channel !== $productValue->getScope()) {
+            return false;
         }
 
-        return false;
+        if (null !== $productValue->getLocale() && $locale !== $productValue->getLocale()) {
+            return false;
+        }
+
+        if ($productValue->getAttribute()->isLocaleSpecific() &&
+            !$productValue->getAttribute()->hasLocaleSpecific($locale)
+        ) {
+            return false;
+        }
+
+        return true;
```

Comment:
Extra question: should we need `addProductValueChecker`? Should we inject them in the constructor?